src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+/** Deal with taking in user input
+ * and making sense of the input
+ *
+ */
+public class Parser {
+    private TaskList tLst;
+
+    Parser(TaskList tLst) {
+        this.tLst = tLst;
+    }
+
+    /**
+     * Reads user input and determines
+     * what action to carry out
+     * @throws DukeException Thrown if input does not make sense
+     * @throws IOException For interrupted I/O operations
+     */
+    public void read() throws DukeException, IOException {
+        Scanner help = new Scanner(System.in);
+        while (true) {
+            String str = help.nextLine();
+            String first = str.split(" ")[0];
+
+            if (first.equals("bye")) {
+                System.out.println("Bye see you again buddy !");
+                break;
+            } else if (first.equals("list")) {
+                tLst.list();
+            } else if (first.equals("mark")) {
+                Task current = tLst.get(Integer.parseInt(str.split(" ")[1]) - 1);
+                tLst.mark(current);
+                Storage.writeToFile(tLst.getTasks());
+
+            } else if (first.equals("unmark")) {
+                Task current = tLst.get(Integer.parseInt(str.split(" ")[1]) - 1);
+                tLst.unmark(current);
+                Storage.writeToFile(tLst.getTasks());
+            } else if (first.equals("delete")) {
+                Task current = tLst.get(Integer.parseInt(str.split(" ")[1]) - 1);
+                tLst.delete(current);
+
+                Storage.writeToFile(tLst.getTasks());
+            } else if (first.equals("find")) {
+                tLst.find(str.split(" ")[1]);
+            } else {
+                if (first.equals("deadline")) {
+                    if (str.endsWith("deadline")) {
+                        throw new DukeException("Oops The description of deadline cannot be empty !");
+                    }
+                    String currD = str.substring(str.indexOf("deadline") + 8, str.indexOf("/by"));
+                    tLst.add(new Deadline(currD.substring(1), str.substring(str.indexOf("/by") + 4)));
```

Comment:
Maybe you could consider breaking the logic up in these two lines of code into smaller, separate lines of code? 

I also found it a little bit difficult to figure out what the 8 and 4 represents. Maybe creating and assigning them to constants might help to add meaning to these arbitrary numbers.